pkg/crypto/tinkcrypto/primitive/composite/ecdhes/ecdhes.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package ecdhes provides implementations of payload encryption using ECDH-ES KW key wrapping with AEAD primitives.
+//
+// The functionality of ecdhes Encryption is represented as a pair of
+// primitives (interfaces):
+//
+//  * ECDHESEncrypt for encryption of data and aad for a given list of recipients keys
```

Comment:
To double check, will this solutions support recipients that each have a different key type (e.g.,  recipient 1 having X25519, recipient 2 having P256, recipient 3 having P384).

Reply:
They can be different types as long as they support the same primitive (CompositeEncrypt/Decrypt interfaces), theoretically. This is how Tink works, as long as the key supports the primitive, it can be used. 

But for key wrapping, we will have to make sure we can support different key types. It's to be discovered when we introduced more key templates to the primitive.